### DISCUSSION TITLE
How to view the hours of every time i've been working on a task during a day?

### DISCUSSION THREAD

**Discussion by TzviRonen** [super-productivity/super-productivity#1434] [Q&A]
Hello
Great App! Thank you for creating it.

Is there a way to view the start and end time of every time I worked on a task?

I have a very long task that seperating it to sub tasks isn't a good idea and i would like to tell how much time i worked on every working session i hade during the same day. When I start a working session on a task i use the "start tracking time" button to start tracking.

**Comment by johannesjo** [comment]
Hey there! Thanks for your question (and I am glad to hear you like the app ;))! 

Unfortunately the answer is no. To keep the model and data usage small the model is a bit simplified in the sense that it only records task creation time, the tasks done time and the amount of time spent on different days, but not detailed session data (which would become big quite fast). As you mention, if you need this kind of information you can use sub tasks.
Ultimately this is a trade-off decision:

A. Have more detailed time data
B. Better performance, smaller and simpler model (=> faster sync, => allows for sync via dropbox which wouldn't be possible like this with all the data)

**Reply by HoroTW** [reply]
Hey there!
I also really enjoy the App!
But I also miss more detailed information about the start and end times....
I would like to see how much time I spent and when... I can imagine a calendar like view of the tasks that where worked on.
Another that would become possible would be to see when you where the most productive in the day.
But such a view/analysis would need the timestamps...
I estimated that in my case it would only take up around 1MB per year - which wouldn't be too bad? Or am I missing something?

I think about trying to add this functionality but I have no experience with TS nor do I know if you would like this change..
Maybe this change would be too fundamental?

**Reply by johannesjo** [reply]
Thanks for your input. A calendar like view already exists imho with the the worklog and the quick history.  But that is not what you are looking for?

**Reply by HoroTW** [reply]
Yeah, calendar like view was not perfect phrased ^^
Maybe more like a day view in which you could see the breaks so that you could see the pause times and the "work efficiency".

![image](https://user-images.githubusercontent.com/8217773/129785783-9db4e5a0-bafe-4a17-bc1c-4bd017094db4.png)
Maybe something like this ^^

**Reply by hpza** [reply]
Hi, I'm just exploring the tool, it loks very nice. This coud be the perfect tool for me except, that I need to now the start and end time of my work session on every task.
What about implementing a webhook which would be triggered on start and end of every session? This could be a tradeoff and offers possibility to integrate with other tools?
Thanks

**Reply by johannesjo** [reply]
I would be open about supplying a more general AI for super productivity. We could stream all redux actions performed via a websocket for example:
https://github.com/johannesjo/super-productivity/issues/3515

Then people could write their own implementation of whatever they need.

**Comment by Lobodiurno** [comment]
This would be a great feature.

**Comment by johannesjo** [comment]
@kiausch maybe also interesting to you.

[a comment by an automated account is not shown]
